api/envoy/extensions/filters/http/sxg/v3alpha/sxg.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.sxg.v3alpha;
+
+import "envoy/extensions/transport_sockets/tls/v3/secret.proto";
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.sxg.v3alpha";
+option java_outer_classname = "SxgProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Signed HTTP Exchange Filter]
+// SXG :ref:`configuration overview <config_http_filters_sxg>`.
+// [#extension: envoy.filters.http.sxg]
+
+// [#next-free-field: 10]
+message SXG {
+  // The SDS configuration for the public key data for the SSL certificate that will be used to sign the
+  // SXG response.
+  transport_sockets.tls.v3.SdsSecretConfig certificate = 1;
```

Comment:
Does the cert info need to come from SDS?  We've recently been adding other options to `CommonTlsContext`, such as the `CertificateProviderInstance` hooks.  Can we leverage that here somehow instead of hard-coding use of SDS?

Reply:
No, it does not need to be SDS. The version of this that we are running currently, uses an internal platform for secrets. I was using the Oauth2 filter extension as a template, and it uses SDS. Happy to use `CertificateProviderInstance` or similar if that's preferred. Do you have an example I could look at?

Reply:
What's the rationale for departing from SDS?

Reply:
@envoyproxy/api-shepherds to get some guidance on SDS vs less SDS for similar use cases.

Reply:
Unfortunately, the way the protos are currently set up, we don't seem to have a single reusable proto that represents a "certificate source" that can be configured for any of the various supported mechanisms.  The infrastructure here is a little weak.

For now, I guess this is okay as-is, just because we don't have anything better to offer.  Just keep in mind that we will probably need to update this in the future as this infrastructure improves.